### PR TITLE
code and headder reorganisation.

### PR DESCRIPTION
**Pull request by JamesCordell**
Hi

Does this make sense? This seems to make the dependencies less complex. I am trying to make this decode a stream from a /dev/ttyS0 so I can read the data over a serial in real time.

### CODE REVIEW

**Comment by mikeller** [comment]
Thanks for your contributtion, @JamesCordell!

File: src/stream.h [hunk 1]
```diff
@@ -26,9 +26,9 @@ mmapStream_t* streamCreate(int fd);
 void streamDestroy(mmapStream_t *stream);
 
 int streamPeekChar(mmapStream_t *stream);
-int streamReadChar(mmapStream_t *stream);
+char streamReadChar(mmapStream_t *stream);
 int streamReadByte(mmapStream_t *stream);
-void streamUnreadChar(mmapStream_t *stream, int c);
+void streamUnreadChar(mmapStream_t *stream);
```
Comment (AndersHoglund) [inline src/stream.h]: Incomplete/Inconstant change. Build fails:
```
%% stream.c
.//src/stream.c:91:6: error: conflicting types for âstreamUnreadCharâ
 void streamUnreadChar(mmapStream_t *stream, int c)
      ^
In file included from .//src/stream.c:11:0:
.//src/stream.h:31:6: note: previous declaration of âstreamUnreadCharâ was here
 void streamUnreadChar(mmapStream_t *stream);
      ^
Makefile:123: recipe for target 'obj/stream.o' failed
make: *** [obj/stream.o] Error 1
```

CI-build fails: http://andwho.sytes.net:8080/job/Borisb_Blackbox_Tools/8/console

Comment (JamesCordell) [reply src/stream.h]: Im still getting use to git an missed this sorry. I see you made the correction should I delete my pull request?

Comment (AndersHoglund) [reply src/stream.h]: No, I have not made any corrections. Please keep your PR that fixes this.

**Pull request merged**